hypervisor/src/hypervisor.rs

```diff
@@ -118,3 +118,21 @@ pub trait Hypervisor: Send + Sync {
     #[cfg(feature = "tdx")]
     fn tdx_capabilities(&self) -> Result<TdxCapabilities>;
 }
+///
+/// Generic MemoryRegion struct
+///
+#[derive(Debug, Default, PartialEq)]
+pub struct UserMemoryRegion {
+    pub slot: u32,
+    pub guest_phys_addr: u64,
+    pub memory_size: u64,
+    pub userspace_addr: u64,
+    pub flags: u32,
+}
+
+pub mod user_memory_region_flags {
+    pub const READ: u32 = 1;
+    pub const WRITE: u32 = 2;
+    pub const EXECUTE: u32 = 4;
+    pub const LOG_DIRTY_PAGES: u32 = 8;
```

Comment:
It is usually ideal to use bit shift operator in this case. 1 << 2, 1 << 3, 1 << 4 when defining values in power of 2.